### DISCUSSION TITLE
Login trigger for normal users

### DISCUSSION THREAD

**Discussion by mezuky04** [drakkan/sftpgo#1375] [General]
Hello. Is it possible to have events triggered when a user logs in (similar to the `Identity provider login` trigger, but for normal users)? Thanks :)

**Comment by drakkan** [comment]
Do you mean the [pre-login hook](https://github.com/drakkan/sftpgo/blob/main/docs/dynamic-user-mod.md)?

**Comment by mezuky04** [comment]
That's exactly what I was looking for. Thanks for taking the time to answer 🙂

**Reply by drakkan** [reply]
No problem. This part

> The JSON response can include only the fields to update instead of the full user.

will only work for very simple use cases. I removed the same thing from the REST API, I cannot find the exact issue now but this mode will create issues with sub-struct and nested objects.
